students/GKim/lesson04/mailroom2.py

```diff
@@ -0,0 +1,208 @@
+
+from textwrap import dedent 
+import sys, os
+
+
+main_donors = [
+          {"name": "Luke Skywalker", "donations": [100.25, 200.55, 50]},
+          {"name": "Han Solo", "donations": [100.80, 50.99, 600]},
+          {"name": "Yoda", "donations": [1000.01, 50, 600.55, 200.47]},
+          {"name": "Ben Kenobe", "donations": [101.32, 500, 60.34]},
+]
+
+def clear_screen():
+    """
+    clears the command screen
+    """
+    os.system("cls" if os.name == "nt" else "clear")
+
+
+def show_list():
+    """
+    shows the donors in the list with donation amounts
+    \n"""
+    print("#" * 25, "The Current Donor List", "#" * 25)
+    index = 1
+    for row in main_donors:
+        print("{:<1}: {:>15}{:>10}: {:<20}".format(index, row["name"], "Amt", str(row["donations"]).replace("[","").replace("]","")))
+        index += 1
+    print("#" * 75 + "\n")
+    return show_list
+
+def main_menu():
+    """
+    Main menu options
+    """
+    prompt = input("\n".join(("\nWelcome to the Mailroom!",
+"Please choose from one of the options below:\n",
+"1: Send a Thank You",
+"2: Create a Report",
+"3: Send to All",
+"4: Quit",
+">>> ")))
+    return prompt
+
+def menu_thank_you():
+    """
+    Menu to user to enter options for thank you
+    """
+    print("""\nTo whom would you like to send a Thank You?
+Please enter full name""")
+    thanks_answer = input("""
+Enter 'LIST' to see Donor list or 'MENU' to return to Main Menu.
+>>> """)
+    return thanks_answer
+
+
+def send_email(name, amount):
+    """
+    This functions sends a letter to an individual donor
+    """
+    print("""\n
+    Dear {n},
+       Thank you for your generous donation of ${a}! This will help our cause
+       immensely in our battle against the darkside.  You, {n} , have made a big 
+       difference in our efforts and we greatly appreciate you!!!  Your ${a} will be 
+       put to great use to our forces against evil!  
+
+       Thank you,
+
+       The FORCE
+    
+    \n""".format(n = name, a = amount))
+
+def send_letter(name, amount):
+    """
+    This function can send a letter to all but mainly to write to file
+    """
+    return dedent(''' 
+          Dear {},
+            Thank you for your very kind donations totaling ${:,.2f}.
+          It will be put to very good use.
+
+                         Sincerely,
+                            -The Force
+          '''.format(name, amount))
+   
+
+def send_thank_you():
+    """
+    Either adds to list or adds a new donor to the list and  a 
+    new donation amount 
+    """
+    clear_screen()
+    while True:
+        thanks_answer = menu_thank_you().strip()
+        if thanks_answer.upper() == "LIST":
+            clear_screen()
+            show_list()
+        elif thanks_answer.upper() == "MENU":
+            clear_screen()
+            break
+        else:
+            idx = len(main_donors)-1
+            in_main_donors = False
+            for x in range(len(main_donors)-1, -1, -1):
```

Comment:
Since we are working with a dict now, you don't need to use a for loop to iterate through it. You can just check if the key exists (where the key would be the name)